File: disposizioni/disposizioni.py
# n! / (n-k)! con:
# - n numero di oggetti 
# - k numero dei posti

# 4! / (4 - 4)! => 4! / 0! => 4*3*2*1 / 1 => 24

def fattoriale(n):
  if (n<=1):
    return 1
  else:
    return n*fattoriale(n-1)


# funzione
def disposizioni(n,k):
  risultato=fattoriale(n)/fattoriale(n-k)
  return risultato

  # __main__
val1 = int(input("Inserisci il numero degli elementi: "))
val2 = int(input("Inserisci il numero dei posti: "))

ris = disposizioni(val1, val2)

print("Il risultato e': " + str(ris))

File: disposizioni/disposizioni.c
#include <stdio.h>

// Funzione fattoriale che calcola il fattoriale di un numero
int fattoriale(int n){
  if (n <= 1){
    return 1;
  } else {
    return n * fattoriale(n - 1);
  }
}

// Funzione disposizioni che calcola il numero di disposizioni
int disposizioni(int n, int k){
  int risultato = fattoriale(n) / fattoriale(n-k);
  return risultato;
}

// Funzione main
int main(){
  int val1, val2, ris;

  printf("Inserisci il numero degli elementi: ");
  scanf("%d", &val1);

  printf("Inserisci il numero dei posti: ");
  scanf("%d", &val2);

  ris = disposizioni(val1, val2);

  printf("Il risultato e': %d\n", ris);

  return 0;
}
